packages/ui-dom-utils/src/ownerDocument.ts

```diff
@@ -36,7 +36,13 @@ import React from 'react'
  * @returns { Document } the owner document
  */
 function ownerDocument(
-  el?: Node | Window | React.ReactElement | ((...args: any[]) => any) | null
+  el?:
+    | Node
+    | Window
+    | React.ReactElement
+    | React.Component
+    | ((...args: any[]) => any)
```

Comment:
I think this function should return `Node | Window | null | undefined` too. Also as I see in `findDOMNode` it has actually 0 arguments, so the signature should be `(() => Node | Window | null | undefined)`

Reply:
I ended up making a new `UIElement` type in the shared-types package and used that everywhere where relevant.